openmcyclus/DepleteReactor.py

```diff
@@ -8,8 +11,603 @@ class DepleteReactor(Facility):
+
+        If it's the end of a cycle, that is recorded. If it is 
+        after a cycle ends, and fuel has not been discharged, 
+        then the fuel is discharged. If it's after a cycle ends, then 
+        fuel is loaded
+        '''
+        print("time:", self.context.time, "tick")
+        if self.retired():
+            print("time:", self.context.time, "retired")
+        #    self.record("RETIRED", "")
+            if self.context.time == self.exit_time + 1:
+                if self.decom_transmute_all == 1:
+                    self.transmute(math.ceil(self.n_assem_core))
+                else:
+                    self.transmute(math.ceil(self.n_assem_core/2))
+
+            while self.core.count > 0:
+                if self.discharge() == False:
+                    break
+            print("time:", self.context.time, "end discharge loop")
+            while (self.fresh_fuel.count > 0) and (self.spent_fuel.space >= self.assem_size):
+                self.spent_fuel.push(self.fresh_fuel.pop())
+
+            print("time:", self.context.time, "decommission?", self.check_decommission_condition())
+            if self.check_decommission_condition():
+                self.decommission()
+                
+
+        print("time:", self.context.time, "end retired loop")
+        if self.cycle_step == self.cycle_time:   
+            print("time:", self.context.time, "transmute", math.ceil(self.n_assem_batch))     
+            self.transmute(math.ceil(self.n_assem_batch))
 
+        if (self.cycle_step >= self.cycle_time) and (self.discharged == False):
+            self.discharged = self.discharge()
+
+        if self.cycle_step >= self.cycle_time:
+            print("time:", self.context.time, "load")
+            print("time:", self.context.time, self.core.count)
+            print("time:", self.context.time, self.fresh_fuel.count)
+            self.load()
+
+        #lib.Logger('5', str("DepleteReactor" + str(self.power_cap) + "is ticking"))
+        print("time:", self.context.time, "end tick")
+        return
+ 
     def tock(self):
-        print("World \n")
+        '''
+        Logic to implement at the tock phase of each 
+        time step. 
+
+        If the prototype is retired, then nothing happens. 
+
+        If it's after a cycle ends and the refueling time has passed,
+        the core is full, and fuel has been discharged, then 
+        the discharged variable is changed to false and the 
+        cycle length counter is restarted.
+
+        If it's the beginning of a new cycle and the core is full, 
+        then a cycle start is recorded. 
+
+        If it's in the middle of a cycle and the core is full, the 
+        the power_cap value is recorded as power generated. If these 
+        conditions aren't met, then a power of 0 is recorded. 
+
+        If it's in the middle of a cycle or the core is full, then 
+        the cycle duration counter increases by one. 
```

Comment:
I like the way you've set up these doc strings

Reply:
Thanks! It's mostly me trying to keep track of actions as I translated things from C++ to python